cupy/core/raw.pyx

```diff
@@ -223,6 +234,27 @@ cdef class RawKernel:
         attr = driver.CU_FUNC_ATTRIBUTE_PREFERRED_SHARED_MEMORY_CARVEOUT
         driver.funcSetAttribute(self.kernel.ptr, attr, fraction)
 
+    def compile(self, options=None, backend=None, log_stream=None):
```

Comment:
Are these options (`options`, `backend`) needed? I think just recreating the RawKernel is enough for users.

Reply:
I think the idea is to compile (or force recompile) an existing `RawKernel` instance so that we can support a development cycle in a console like this:
```python
ker = RawKernel(...)  
ker.compile(options=some_options, log_stream=sys.stdout)
ker((16,), (128,), args)   # test
ker.compile(options=some_other_options, log_stream=sys.stdout)
ker((16,), (128,), args)   # test